pkg/grpc/server.go

```diff
@@ -179,14 +187,19 @@ func (s *server) StartNonBlocking() error {
 }
 
 func (s *server) Close() error {
+	defer s.wg.Wait()
+	if s.closed.CompareAndSwap(false, true) {
+		close(s.closeCh)
+	}
+
 	for _, server := range s.servers {
 		// This calls `Close()` on the underlying listener.
-		server.GracefulStop()
+		go server.GracefulStop()
 	}
 
 	if s.api != nil {
 		if closer, ok := s.api.(io.Closer); ok {
-			closer.Close()
+			return closer.Close()
```

Comment:
I would add an error check like "if err != nil"...

I can see a day when someone adds something to the Close() method after this, and doesn't realize that there's a `return` above.

Reply:
Yes, good shout